### DISCUSSION TITLE
Execute method of IJob getting called multiple times

### DISCUSSION THREAD

**Discussion by viveknuna** [quartznet/quartznet#2055] [Q&A]
I am using Quartz 3.5.1 version. There is a class which is implementing the IJob interface. It shows 0 references for the Execute method, but it is getting called multiple times. I have no clue where its getting called. Could you please provide some info on this?

![image](https://github.com/quartznet/quartznet/assets/3524495/5182a145-befe-4f60-8815-99e17dc8eabe)

**Comment by viveknuna** [comment]
@lahma nobody responds. I don’t know what kind of discussion it is.

what is the best way to get help on questions related to Quartz?

**Comment by jafin** [comment]
@viveknuna I think you need to appreciate that forums/open source etc are best effort unpaid offering.    Just because you don't receive an answer in 11 hours doesn't mean no discussion occurs. people have jobs, different timezones, and priorities.    

Anyhow,  have you read https://www.quartz-scheduler.net/documentation/quartz-3.x/tutorial/more-about-jobs.html
It should explain how Execute is triggered or at least provide a better understanding.

**Reply by viveknuna** [reply]
@jafin I also believe in forums and open channels. that is why I am also a contributor on StackOverflow with a significantly lower reputation https://stackoverflow.com/users/6527049/vivek-nuna?tab=profile.

I said so because I posted a few questions 2-3 days back but have not received any response.

So I thought there might be a better way to get support, no intention to criticize this channel.

**Comment by nikodojan** [comment]
I have similar problem. I am in testing phase, so I have scheduled only a few jobs. I am working without database, only in-memory. I have one scheduler and two jobs with each one trigger with cron schedule, set to run every 10 minutes. For some reason, when the job is fired, it is scheduled multiple times, with 1 second delay. I am logging previous, current und next execution time, which are 1 seconds apart. 
I don't quite understand why that happens. I already tried to disable concurrency, ignore misfiring, checked for multiple schedulers or Start() calls. Still, local as well as in Azure, the jobs are fired multiple times at a time.
Any suggestions? 

```
+	StartTimeUtc        {01/08/2025 13.30.05 +00:00}
+	TimeZone        {(UTC+01:00) Brussels, Copenhagen, Madrid, Paris}
+	cronEx        {* */5 * ? * *}
	misfireInstruction        -1
+	nextFireTimeUtc        {01/08/2025 13.30.06 +00:00}
+	previousFireTimeUtc        {01/08/2025 13.30.05 +00:00}
+	startTimeUtc (Quartz.Impl.Triggers.AbstractTrigger)        {01/01/0001 00.00.00 +00:00}
+	startTimeUtc        {01/08/2025 13.30.05 +00:00}
```

**Reply by lahma** [reply]
Did you try to enable logging and reading Quartz's decisions?

**Reply by nikodojan** [reply]
Thanks for the reply.
I am logging, and have tried to set Quartz logging down to trace, but it doesn't tell me much. I tried different tips from the docs, handling all exceptions (none thrown though), prevent concurrent execution. Tried different cron schedules, all with minutes though. It seems only to do it with jobs that call on other functions. If I simply log something to console as a test job, then it works fine.

```
dbug: Quartz.Core.JobRunShell[0]
      Trigger instruction : NoInstruction
dbug: Quartz.Core.QuartzSchedulerThread[0]
      Batch acquisition of 1 triggers
dbug: Quartz.Core.QuartzSchedulerThread[0]
      Batch acquisition of 1 triggers
dbug: Quartz.Core.JobRunShell[0]
      Calling Execute on job DEFAULT.CleanupTimer-job

(my own logs in from the job using the execution context)
info: CleanupTimer[0]
      03/08/2025 13.44.19 +00:00 <- last triggered
info: CleanupTimer[0]
      03/08/2025 13.44.18 +00:00 <- current
info: CleanupTimer[0]
      03/08/2025 13.44.20 +00:00 <- next scheduled
info: CleanupTimer[0]
      [08/03/2025 13:44:19 +00:00] CleanupTimer executed. Trigger: DEFAULTCleanupTimer-trigger, FireInstanceId: 638898254422776717
```

Is there more logging available?

**Reply by lahma** [reply]
It's all guesswork, maybe create a code repository which shows the erratic behavior, it's impossible to guess.

**Reply by nikodojan** [reply]
Found the issue. As often, it was sitting ca. 45cm in front of the screen.
I misunderstood the meaning of the cron expression, I wrote. So, Quartz scheduled exactly as the cron expression states, every second `"* */5 * ? * *"`. What I actually is wanted is to fire once every 5 minutes and have to write `"0 */5 * ? * *"` instead.
Sorry for the confusion and thanks for your responses.

**Comment by Sicos1977** [comment]
PEBCAK = Problem Exists Between Chair And Keyboard
Carbon based error (what life is made of) :-)
